src/main/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringcreateanorderapi/service/DeviceWearerAddressService.kt

```diff
@@ -0,0 +1,67 @@
+package uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.service
+
+import jakarta.persistence.EntityNotFoundException
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.models.DeviceWearerAddress
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.models.enums.DeviceWearerAddressType
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.models.enums.FormStatus
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.repository.DeviceWearerAddressRepository
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.repository.OrderFormRepository
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.resource.UpdateDeviceWearerAddressDto
+import java.util.UUID
+
+@Service
+class DeviceWearerAddressService(
+  val orderRepo: OrderFormRepository,
+  val addressRepo: DeviceWearerAddressRepository,
+) {
+  fun getAddress(
+    orderId: UUID,
+    username: String,
+    addressType: DeviceWearerAddressType,
+  ): DeviceWearerAddress {
+    // Verify the order belongs to the user and is in draft state
+    val order = orderRepo.findByIdAndUsernameAndStatus(
+      orderId,
+      username,
+      FormStatus.IN_PROGRESS,
+    ).orElseThrow {
+      EntityNotFoundException("Order with id $orderId does not exist")
```

Comment:
This feels bit wrong, I understand you only want to retrieve addresses that is editable, but the message suggest the order not exists

Reply:
True, I can fix that